zio-query/shared/src/main/scala/zio/query/CompletedRequestMap.scala

```diff
@@ -73,29 +100,58 @@ sealed abstract class CompletedRequestMap { self =>
   final def requests: Set[Request[_, _]] =
     map.keySet.asInstanceOf[Set[Request[_, _]]]
 
-  /**
-   * Whether the completed requests map is empty.
-   */
-  final def isEmpty: Boolean =
-    map.isEmpty
-
-  final private[query] def toMutableMap: mutable.HashMap[Request[?, ?], Exit[Any, Any]] =
-    mutable.HashMap.from(map.asInstanceOf[collection.Map[Request[?, ?], Exit[Any, Any]]])
-
   final override def toString: String =
     s"CompletedRequestMap(${map.mkString(", ")})"
+
+  final private[query] def underlying: collection.Map[Request[?, ?], Exit[Any, Any]] =
+    map.asInstanceOf[collection.Map[Request[?, ?], Exit[Any, Any]]]
 }
 
 object CompletedRequestMap {
 
   val empty: CompletedRequestMap =
     new Immutable(immutable.HashMap.empty)
 
+  /**
+   * Combines all completed request maps into a single one.
+   *
+   * This method is left-associated, meaning that if a request is present in
+   * multiple maps, it will be overriden by the last map in the list.
+   */
+  def fromCompletedRequestMaps(maps: Iterable[CompletedRequestMap]): CompletedRequestMap = {
```

Comment:
Should I rename this to `combine` perhaps?

Reply:
yeah